runtime/src/qio/regexp/re2/re2-interface.cc

```diff
@@ -152,10 +121,10 @@ void re_free(re_t* re)
   delete re;
 }
 
-
 static
 re_t* local_cache_get(const char* str, int64_t str_len, const qio_regexp_options_t* options) {
-  re_cache* c = local_cache();
+  thread_local re_cache cache;
```

Comment:
I haven't worked with C++ `thread_local` before. But it would be more intuitive to me if this declaration were `static` and outside of the function. Does that make sense to do in this case?

Reply:
Ah, looking at some documentation I see that the difference is when the thread-local storage is initialized. The way it is written here as a local variable, it's initialized when `local_cache_get` runs for the first time. Vs. if it is a global it will be initialized at program startup. I don't know if there's a performance difference between these two but I think either is reasonable and it makes sense to initialize the cache only if regular expression operations are used.

Reply:
Yes, I thought about that initialization time (either at thread startup or at function entry), and thought it should only be initialized if the function is called, so that we don't impose that cost on all threads regardless if they use regexp or not.